ffyaml/yaml.go

```diff
@@ -0,0 +1,22 @@
+package ffyaml
+
+import (
+	"fmt"
+	"io"
+
+	"gopkg.in/yaml.v2"
+)
+
+func Parser(r io.Reader, set func(name, value string) error) error {
+	var m map[string]string
+	d := yaml.NewDecoder(r)
+	if err := d.Decode(&m); err != nil && err.Error() != "EOF" {
```

Comment:
You should be able to check for io.EOF directly here, e.g.

```go
if err := d.Decode(&m); err != nil && err != io.EOF {
```

Reply:
Also, do you know how to trigger [the branch in Decode that returns io.EOF](https://github.com/go-yaml/yaml/blob/51d6538a90f86fe93ac480b35f37b2be17fef232/yaml.go#L122-L125)? It would be good to add a test case for that.